product_subscription/models/subscription.py

```diff
@@ -288,6 +303,31 @@ def create(self, vals):
 
         return super(SubscriptionObject, self).create(vals)
 
+    @api.multi
+    @api.depends('subscribed_on', 'template')
+    def _compute_date_end(self):
+        for subscription in self:
+            if not subscription.subscribed_on:
+                return False
+
+            elif subscription.template.time_span_unit == 'year':
+                subscription.end_date = add_years(
```

Comment:
be careful when computing the subscription end date on the subscription_on field. This field give the first subscription date. It doesn't give the start date of the ongoing subscription which could have been renewed. It that case the end date will be wrong.

I propose to add a start date which will be the same than the subscribed_on for a new subscription and the effective payment date for a renew.

Reply:
Good comment but the subscriptions are not renewed anymore. Every new request is a new subscription. It is therefore always the subscription date.

Reply:
This is madness !

https://tenor.com/view/gerald-butler-madness-mad-angry-insane-gif-7844903